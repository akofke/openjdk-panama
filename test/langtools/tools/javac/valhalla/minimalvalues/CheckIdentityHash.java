/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support identityHashCode
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckIdentityHash.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values  CheckIdentityHash.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckIdentityHash {
    int identityHashCode(CheckIdentityHash x) {
        return 0;
    }
    void test(CheckIdentityHash v) {
        this.identityHashCode(v);      // <- ok
        System.identityHashCode(v);    // <- error
        System.identityHashCode(this); // <- error
        java.lang.System.identityHashCode(v);    // <- error
        java.lang.System.identityHashCode(this); // <- error
    }
}
