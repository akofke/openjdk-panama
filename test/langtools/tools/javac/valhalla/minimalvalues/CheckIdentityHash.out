- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckIdentityHash.java:14:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash.java:15:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash.java:16:42: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash.java:17:42: compiler.warn.value.does.not.support: identityHashCode
1 error
5 warnings
